test/core/expectations/compound_data/test_expectations_cursor.pkb

```diff
@@ -1041,5 +1041,516 @@ Rows: [ 2 differences ]%
+    l_expected SYS_REFCURSOR;
+    l_actual_message   varchar2(32767);
+    l_expected_message varchar2(32767);
+  begin
+    --Arrange
+    open l_expected for select username, user_id from all_users union all
+    select 'TEST' username, -600 user_id from dual order by 1 desc;
+    
+    open l_actual for select username, user_id from all_users union all
+    select 'TEST' username, -610 user_id from dual order by 1 asc;
+     
+    --Act
+    ut3.ut.expect(l_actual).to_equal(l_expected).join_by('USERNAME');
+    --Assert
+ l_expected_message := q'[%Actual: refcursor [ count = % ] was expected to equal: refcursor [ count = % ]
+%Diff:%
+%Rows: [ 1 differences ]%
+%Expected: <USER_ID>-600</USER_ID> for key: TEST%
+%Actual:   <USER_ID>-610</USER_ID> for key: TEST%]';
+    l_actual_message := ut3.ut_expectation_processor.get_failed_expectations()(1).message;
+    --Assert
+    ut.expect(l_actual_message).to_be_like(l_expected_message);
+  end;
+  
+  procedure cursor_joinby_cmp_twocol_fail is
+    l_actual   SYS_REFCURSOR;
+    l_expected SYS_REFCURSOR;
+    l_actual_message   varchar2(32767);
+    l_expected_message varchar2(32767);
+  begin
+    --Arrange
+    open l_expected for select username, user_id from all_users union all
+    select 'TEST' username, -600 user_id from dual order by 1 desc;
+    
+    open l_actual for select username, user_id from all_users union all
+    select 'TEST' username, -610 user_id from dual order by 1 asc;
+     
+    --Act
+    ut3.ut.expect(l_actual).to_equal(l_expected).join_by(ut3.ut_varchar2_list('USERNAME,USER_ID'));
+    --Assert
+ l_expected_message := q'[%Actual: refcursor [ count = % ] was expected to equal: refcursor [ count = % ]
+%Diff:%
+%Rows: [ 2 differences ]%
+%Missing:  <USERNAME>TEST</USERNAME><USER_ID>-600</USER_ID> for key: -600:TEST%
+%Extra:    <USERNAME>TEST</USERNAME><USER_ID>-610</USER_ID> for key: -610:TEST%]';
+    l_actual_message := ut3.ut_expectation_processor.get_failed_expectations()(1).message;
+    --Assert
+    ut.expect(l_actual_message).to_be_like(l_expected_message);
+  end;
+  
+  procedure cur_joinby_cmp_threcol_fail is
+    l_actual   SYS_REFCURSOR;
+    l_expected SYS_REFCURSOR;
+    l_actual_message   varchar2(32767);
+    l_expected_message varchar2(32767);
+  begin
+    --Arrange
+    open l_expected for select username, user_id,'Y' is_valid from all_users union all
+    select 'TEST' username, -600 user_id,'Y' is_valid from dual order by 1 desc;
+    
+    open l_actual for select username, user_id,'Y' is_valid from all_users union all
+    select 'TEST' username, -610 user_id,'Y' is_valid from dual order by 1 asc;
+     
+    --Act
+    ut3.ut.expect(l_actual).to_equal(l_expected).join_by(ut3.ut_varchar2_list('USERNAME,IS_VALID'));
+    --Assert
+ l_expected_message := q'[%Actual: refcursor [ count = % ] was expected to equal: refcursor [ count = % ]
+%Diff:%
+%Rows: [ 1 differences ]%
+%Expected: <USER_ID>-600</USER_ID> for key: TEST:Y%
```

Comment:
I believe, this could be more readable if it was:
`PK <USERNAME>TEST</USERNAME><IS_VALID>Y</IS_VALID> - Expected: <USER_ID>-600</USER_ID>`
`PK <USERNAME>TEST</USERNAME><IS_VALID>Y</IS_VALID> - Actual:   <USER_ID>-610</USER_ID>`

Reply:
Make sense